lib/libvcc/vcc_backend.c

```diff
@@ -575,6 +583,28 @@ vcc_ParseHostDef(struct vcc *tl, const struct token *t_be, const char *vgcname)
 		Fb(tl, 0, "\"0.0.0.0\"");
 	Fb(tl, 0, ",\n");
 
+	/*
+	 * Emit the authority field, falling back to hosthdr, then host.
+	 *
+	 * When authority is "", sending the TLV is disabled.
+	 *
+	 * Falling back to host may result in an IP address in authority,
+	 * which is an illegal SNI HostName (RFC 4366 ch. 3.1). But we
+	 * document the potential error, rather than try to find out
+	 * whether or not Emit_Sockaddr() had to look up a name.
```

Comment:
We also don't try to enforce that the configured `.authority` or `.hosthdr` attributes have appropriate values, unless I remember wrong.

Maybe the syntax should be explicit?

```vcl
backend foo {
    .via = proxy;
    .authority = "";
}

# vs

backend bar {
    .via = proxy;
    .authority = none;
}
```

I need to check whether we send an empty authority TLV or nothing when we configure an empty string.

Reply:
On the last sentence, it says what it does in the comment which you quoted.

This was relevant if the empty vs. no authority actually made a difference, and I do not think it does. Do you know better?